neptune_optuna/impl/__init__.py

```diff
@@ -92,47 +95,72 @@ class NeptuneCallback:
         >>> import neptune.new.integrations.optuna as optuna_utils
         ... neptune_callback = optuna_utils.NeptuneCallback(run)
 
-        Pass NeptuneCallback to the Optuna Study:
+        Or `optionally` pass a list of objective names:
+            Single objective:
+            ... neptune_callback = optuna_utils.NeptuneCallback(run, target_names=['accuracy'])
+            Multi-objective:
+            ... neptune_callback = optuna_utils.NeptuneCallback(run, target_names=['FLOPS', 'accuracy'])
+
+        Log single and multi-objective Study metadata to Neptune by passing NeptuneCallback to the Optuna Study:
         >>> study = optuna.create_study(direction='maximize')
         ... study.optimize(objective, n_trials=5, callbacks=[neptune_callback])
 
+
     For more information, see `Neptune-Optuna integration docs page`_.
     .. _Neptune Optuna integration docs page:
        https://docs.neptune.ai/integrations-and-supported-tools/hyperparameter-optimization/optuna
     """
 
-    def __init__(self,
-                 run: neptune.Run,
-                 base_namespace: str = '',
-                 plots_update_freq: Union[int, str] = 1,
-                 study_update_freq: Union[int, str] = 1,
-                 visualization_backend: str = 'plotly',
-                 log_plot_contour: bool = True,
-                 log_plot_edf: bool = True,
-                 log_plot_parallel_coordinate: bool = True,
-                 log_plot_param_importances: bool = True,
-                 log_plot_pareto_front: bool = True,
-                 log_plot_slice: bool = True,
-                 log_plot_intermediate_values: bool = True,
-                 log_plot_optimization_history: bool = True):
+    def __init__(
+        self,
+        run: neptune.Run,
+        base_namespace: str = "",
+        target_names: List[str] = None,
```

Comment:
@Blaizzy @kamil-kaczmarek this change is breaking, unfortunately. Any new params should go last, else we break code when people don't use named params only, but rely on positions.

Reply:
yep good catch!

@Blaizzy can you refactor accordingly?

Reply:
@Herudaio 
This PR is closed, you should comment on the https://github.com/neptune-ai/neptune-optuna/pull/14